File: cs/ex2_1_2.cs
abstract class EmailDst {
    public abstract void sendEmail(string content);
}

class Person : EmailDst {
    public override void sendEmail(string content) {}
}

class Example {
    static void sendNewYearEmail(EmailDst dst) {
        dst.sendEmail("");
    }
    
    public static void Main() {
        Person pr = new Person();
        sendNewYearEmail(pr);
    }
}

File: typescript/ex2_1_2.ts
abstract class EmailDst {
    abstract sendEmail(content: string): void;
}

class Person {
    sendEmail(content: string): void {}
}

let pr: Person = new Person();

function sendNewYearEmail(dst: EmailDst): void {
    dst.sendEmail("");
}
sendNewYearEmail(pr);

function sendNewYearEmail2(dst: { sendEmail: (content: string) => void }): void {
    dst.sendEmail("");
}
sendNewYearEmail2(pr);
